### PR TITLE
Add EIP-712 hash display before signing

### PR DESCRIPTION
**Pull request by austinchandra**

### CODE REVIEW

File: ledger.go [hunk 1]
```diff
@@ -154,6 +163,29 @@ func EvmosLedgerDerivation(config params.EncodingConfig) LedgerDerivation {
...
+	fmt.Printf("Signing the following payload with EIP-712:\n")
+	fmt.Printf("- Domain: %v\n", bytesToHexString(domainSeparator))
+	fmt.Printf("- Message: %v\n", bytesToHexString(typedDataHash))
+
+	return nil
+}
+
+func bytesToHexString(bytes []byte) string {
+	return "0x" + strings.ToUpper(hex.EncodeToString(bytes))
+}
```
Comment (fedekunze) [inline ledger.go:187]: You can use the `common.Bytes2Hex` function instead

File: ledger.go [hunk 2]
```diff
@@ -101,6 +103,7 @@ func (e EvmosSECP256K1) GetAddressPubKeySECP256K1(hdPath []uint32, hrp string) (
 }
 
 func (e EvmosSECP256K1) SignSECP256K1(hdPath []uint32, signDocBytes []byte) ([]byte, error) {
+	fmt.Printf("Generating payload, please check your Ledger...\n")
 	if e.primaryWallet == nil {
```
Comment (fedekunze) [inline ledger.go:107]: nit, I'd change the order here

File: ledger.go [hunk 3]
```diff
@@ -135,6 +138,12 @@ func (e EvmosSECP256K1) SignSECP256K1(hdPath []uint32, signDocBytes []byte) ([]b
 		return make([]byte, 0), errors.New(fmt.Sprintf("Could not decode sign doc as either Amino or Protobuf.\n Amino: %v\n Protobuf: %v\n", errAmino, errProtobuf))
 	}
 
+	// Display EIP-712 message hash for user to verify
+	err = e.displayEIP712Hash(typedData)
+	if err != nil {
```
Comment (fedekunze) [inline ledger.go:143]: nit, I suggest using this pattern from now on to keep the `err` in scope
```go
if err := <CONDITION>; err != nil {
   // ...
}
```

File: ledger.go (hunk 1 above)
Comment (austinchandra) [reply ledger.go:187]: Same implementation:

```
// Bytes2Hex returns the hexadecimal encoding of d.
func Bytes2Hex(d []byte) string {
	return hex.EncodeToString(d)
}
```

**Pull request merged**